### ISSUE TITLE
False-positive (E1130) on the return value of `random.sample`

### ISSUE THREAD

**Issue by bersbersbers** [PyCQA/pylint#7706]
### Bug description
```python
"""False-positive invalid-unary-operand-type"""
from random import sample

all_i: list[int] = [0, 1, 2]

for i in all_i:
    print(-i)

for i in sample(all_i, len(all_i)):
    print(-i)  # pylint raises E1130
```

### Command used

```shell
pylint bug.py
```

### Pylint output

```shell
************* Module bug
bug.py:10:10: E1130: bad operand type for unary -: NoneType (invalid-unary-operand-type)
```

### Expected behavior

No error. Related: #811, of which the `invalid-sequence-index` part seems to be unresolved/have regressed.

### Pylint version

```shell
pylint 2.15.5
astroid 2.12.12
Python 3.11.0 (main, Oct 24 2022, 18:26:48) [MSC v.1933 64 bit (AMD64)]
```

### OS / Environment

Windows 10

**Comment by bersbersbers** [comment]
This seems to be fixed with `astroid==2.12.13` / `pylint==2.15.6`. Thanks :)
